src/shared/find-esp.c

```diff
@@ -438,8 +443,8 @@ int find_esp_and_warn(
                 goto found;
         }
 
-        FOREACH_STRING(_path, "/efi", "/boot", "/boot/efi") {
-                path = _path;
+        FOREACH_STRING(p, "/efi", "/boot", "/boot/efi") {
+                path = prefix_roota(root, p);
```

Comment:
I'd say this no alloca rule in loops exists for a reason. So this should either be reworked or add a suppression comment with explanation…

Reply:
The reason is that input-dependent loop length could cause stack exhaustion, but this is static

Reply:
Hence why I said this should have a https://lgtm.com/help/lgtm/alert-suppression then.

Reply:
yep, adding it

Reply:
> this is static

Last time I checked LGTM didn't complain about "static" loops. It complains here because `root` is passed via `--arg-root` and isn't limited anywhere. I can imagine scenarios where `ARG_MAX` * 3 can overflow the stack so if this is supposed to stay I don't think the warning should be suppressed.

Reply:
Well, if there is a chance for this to blow the stack, it should certainly be changed.

Reply:
It's a false positive because of the fixed loop: "1 for Call to alloca in a **loop**"

Reply:
I wonder where exactly arg_root is bound?